### PR TITLE
Split up comment printing

### PR DESCRIPTION
**Pull request by azz**
Unblocks https://github.com/prettier/prettier-python/pull/9

~This _works_ but the implementation leaves a bit to be desired. Need to improve the way the `printer`, `parser` and `plugins` are passed around rather than being looked up each time or added as more args everywhere.~

I've put `astFormat`, `plugins` and `printer` into the `options` object in the `normalize()` function. Not sure if this is the best solution but it cleans up the code significantly.

### CODE REVIEW

File: src/language-js/printer-estree.js [hunk 1]
```diff
@@ -5167,9 +5167,77 @@ function willPrintOwnComments(path) {
   );
 }
 
+function whatShouldICallThis(node) {
```
Comment (azz) [inline src/language-js/printer-estree.js]: Seriously, no idea what to call this function...

Comment (azz) [reply src/language-js/printer-estree.js]: Going with `canAttachComment`.

**Comment by vjeux** [comment]
I was going to suggest canAttachComment as well :)

File: index.js [hunk 2]
```diff
@@ -179,16 +184,21 @@ function findSiblingAncestors(startNodeAndParents, endNodeAndParents) {
...
+function findNodeAtOffset(node, offset, options, predicate, parentNodes) {
   predicate = predicate || (() => true);
   parentNodes = parentNodes || [];
   const start = util.locStart(node);
   const end = util.locEnd(node);
   if (start <= offset && offset <= end) {
-    for (const childNode of comments.getSortedChildNodes(node)) {
+    for (const childNode of comments.getSortedChildNodes(
+      node,
+      undefined,
```
Comment (vjeux) [inline index.js]: nit: put an inline comment with that this variable is

**Pull request merged**